src/prefect/tasks.py

```diff
@@ -310,7 +310,7 @@ def __init__(
             Callable[["TaskRunContext", Dict[str, Any]], Optional[str]]
         ] = None,
         cache_expiration: Optional[datetime.timedelta] = None,
-        task_run_name: Optional[Union[Callable[[], str], str]] = None,
+        task_run_name: Optional[Union[Callable[..., str], str]] = None,
```

Comment:
Is it possible to type the `Callable` more specifically? That way users will know `parameters` is available via autocomplete.

Reply:
yeah i think we can, in [8dfcc06](https://github.com/PrefectHQ/prefect/pull/15773/commits/8dfcc061c3da84d32e48052daea8930fee8a77ee) i said that for now its a `Callable[[], str] | Callable[[dict], str]`

though users should mostly control their destinies since we only check that there's a kwarg called `parameters` and pass it on, they can annotate that however they want